### ISSUE TITLE
Test CentOS 7.3 install

### ISSUE THREAD

**Issue by erxspin** [fusionpbx/fusionpbx-install.sh#88]
``$ yum -y install wget``

``$ wget https://raw.githubusercontent.com/fusionpbx/fusionpbx-install.sh/master/install.sh -O install.sh && sh install.sh``

...
Enabled color support
CentOS
Detected CentOS
Updating system before starting.
...
Fetching Installer
Cloning into 'fusionpbx-install.sh'...
remote: Counting objects: 1797, done.
remote: Compressing objects: 100% (14/14), done.
remote: Total 1797 (delta 3), reused 0 (delta 0), pack-reused 1783
Receiving objects: 100% (1797/1797), 229.08 KiB | 93.00 KiB/s, done.
Resolving deltas: 100% (717/717), done.
install.sh: line 184: ./install.sh: Permission denied

``$ lsb_release -a``

LSB Version:    :core-4.1-amd64:core-4.1-noarch
Distributor ID: CentOS
Description:    CentOS Linux release 7.3.1611 (Core)
Release:        7.3.1611
Codename:       Core

``$ uname -a``

Linux host01.localdomain 3.10.0-514.10.2.el7.x86_64 #1 SMP Fri Mar 3 00:04:05 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux

``$ ls -la /etc/*release``

-rw-r--r--. 1 root root  38 Nov 30 00:12 /etc/centos-release
-rw-r--r--. 1 root root 393 Nov 30 00:12 /etc/os-release
lrwxrwxrwx. 1 root root  14 Mar 14 12:42 /etc/redhat-release -> centos-release
lrwxrwxrwx. 1 root root  14 Mar 14 12:42 /etc/system-release -> centos-release

``$ cat /etc/*release``

``$ cat /etc/centos-release``

CentOS Linux release 7.3.1611 (Core)

``$ cat /etc/os-release``

NAME="CentOS Linux"
VERSION="7 (Core)"
ID="centos"
ID_LIKE="rhel fedora"
VERSION_ID="7"
PRETTY_NAME="CentOS Linux 7 (Core)"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:centos:centos:7"
HOME_URL="https://www.centos.org/"
BUG_REPORT_URL="https://bugs.centos.org/"

CENTOS_MANTISBT_PROJECT="CentOS-7"
CENTOS_MANTISBT_PROJECT_VERSION="7"
REDHAT_SUPPORT_PRODUCT="centos"
REDHAT_SUPPORT_PRODUCT_VERSION="7"

**Comment by markjcrane** [comment]
Why I used config.sh in debian install script and why Centos install will use it to in the future.
Examples config files from other projects postgresql.conf, nginx.com, fail2ban/jail.conf and php.ini notice the large majority of the settings on all of these are lower case.

Over the long run there will be too many parameters or command line args to the install script so I think install settings in config.sh is a better way to go.

**Comment by erxspin** [comment]
Yes, I agree with you.
I will do it at next step of CentOS code transformation to Debian like code.

What do you think about a common version of config.sh shared among various distributions ?

**Comment by erxspin** [comment]
Log of next cycle of testing under CentOS

``./install.sh``

\...
Disabling SELinux
Reboot required after installation completes
SELinux disabled
Installing FreeSWITCH
...
Installed:
  freeswitch-release.noarch 0:1-6
...
Retrieving key from file:///etc/pki/rpm-gpg/RPM-GPG-KEY-FREESWITCH
Importing GPG key 0x25E010CF:
 Userid     : "FreeSWITCH Package Signing Key <packages@freeswitch.org>"
 Fingerprint: 20b0 6ee6 21ab 150d 40f6 079f d76e dc77 25e0 10cf
 Package    : freeswitch-release-1-6.noarch (installed)
 From       : /etc/pki/rpm-gpg/RPM-GPG-KEY-FREESWITCH
Retrieving key from file:///etc/pki/rpm-gpg/RPM-GPG-KEY-EPEL-7
Importing GPG key 0x352C64E5:
 Userid     : "Fedora EPEL (7) <epel@fedoraproject.org>"
 Fingerprint: 91e9 7d7c 4a5e 96f1 7f3e 888f 6a2f aea2 352c 64e5
 Package    : epel-release-7-9.noarch (@extras)
 From       : /etc/pki/rpm-gpg/RPM-GPG-KEY-EPEL-7
Running transaction check
...
FreeSWITCH installed
Installing FusionPBX
...
Loaded plugins: fastestmirror
http://packages.irontec.com/centos///repodata/repomd.xml: [Errno 14] HTTP Error 404 - Not Found
Trying other mirror.
...
failure: repodata/repomd.xml from irontec: [Errno 256] No more mirrors to try.
http://packages.irontec.com/centos///repodata/repomd.xml: [Errno 14] HTTP Error 404 - Not Found
--2017-03-16 16:35:04--  https://forensics.cert.org/cert-forensics-tools-release-el7.rpm
...
Using version 4.2
Cloning into '/var/www/fusionpbx'...
...
FusionPBX Installed
Installing PostgreSQL 9.4
...
Initalize PostgreSQL database
resources/postgres.sh: line 21: /usr/pgsql-9.4/bin/postgresql94-setup: No such file or directory
sed: can't read /var/lib/pgsql/9.4/data/pg_hba.conf: No such file or directory
sed: can't read /var/lib/pgsql/9.4/data/pg_hba.conf: No such file or directory
Failed to restart postgresql-9.4.service: Unit not found.
sudo: unknown user: postgres
sudo: unable to initialize policy plugin
...
PostgreSQL 9.4 installed
Generating a 2048 bit RSA private key
........................................................................+++
................................+++
writing new private key to '/etc/ssl/private/nginx.key'

\-----
Subject Attribute organizationUnitName has no known NID, skipped
No value provided for Subject Attribute emailAddress, skipped
Installing nginx
...
awk: fatal: cannot open file `/etc/nginx/nginx.conf' for reading (No such file or directory)
sed: can't read /etc/nginx/nginx.conf: No such file or directory
nginx installed
Configuring php/nginx/php-fpm and permissions
sed: can't read /etc/nginx/nginx.conf: No such file or directory
sed: can't read /etc/php.ini: No such file or directory
sed: can't read /etc/php.ini: No such file or directory
sed: can't read /etc/php-fpm.d/www.conf: No such file or directory
sed: can't read /etc/php-fpm.d/www.conf: No such file or directory
sed: can't read /etc/php-fpm.d/www.conf: No such file or directory
sed: can't read /etc/php-fpm.d/www.conf: No such file or directory
sed: can't read /etc/php-fpm.d/www.conf: No such file or directory
php/nginx/php-fpm and permissions configured
Configuring FirewallD
success
success
...
cp: cannot create regular file ▒/etc/fail2ban/filter.d/freeswitch-dos.conf▒: No such file or directory
cp: cannot create regular file ▒/etc/fail2ban/filter.d/freeswitch-ip.conf▒: No such file or directory
cp: cannot create regular file ▒/etc/fail2ban/filter.d/freeswitch-404.conf▒: No such file or directory
cp: cannot create regular file ▒/etc/fail2ban/filter.d/freeswitch.conf▒: No such file or directory
cp: cannot create regular file ▒/etc/fail2ban/filter.d/fusionpbx.conf▒: No such file or directory
cp: cannot create regular file ▒/etc/fail2ban/filter.d/nginx-404.conf▒: No such file or directory
cp: cannot create regular file ▒/etc/fail2ban/filter.d/nginx-dos.conf▒: No such file or directory
cp: cannot create regular file ▒/etc/fail2ban/jail.local▒: No such file or directory
Failed to restart fail2ban.service: Unit not found.
Fail2ban installed
Restarting packages for final configuration
Failed to restart php-fpm.service: Unit not found.
Failed to restart nginx.service: Unit not found.
Failed to restart fail2ban.service: Unit not found.
Restart of service complete
sudo: unknown user: postgres
sudo: unable to initialize policy plugin
sudo: unknown user: postgres
sudo: unable to initialize policy plugin
resources/finish.sh: line 40: php: command not found
resources/finish.sh: line 43: psql: command not found
resources/finish.sh: line 46: php: command not found
resources/finish.sh: line 49: /usr/bin/php: No such file or directory
resources/finish.sh: line 50: /usr/bin/php: No such file or directory
resources/finish.sh: line 53: php: command not found
resources/finish.sh: line 54: psql: command not found
resources/finish.sh: line 57: psql: command not found
resources/finish.sh: line 61: /usr/bin/php: No such file or directory
resources/finish.sh: line 63: psql: command not found
resources/finish.sh: line 75: php: command not found
Created symlink from /etc/systemd/system/wpa_supplicant.service to /dev/null.
Failed to execute operation: No such file or directory
Created symlink from /etc/systemd/system/multi-user.target.wants/ntpd.service to /usr/lib/systemd/system/ntpd.service.
Failed to execute operation: No such file or directory
Failed to execute operation: No such file or directory
Created symlink from /etc/systemd/system/multi-user.target.wants/memcached.service to /usr/lib/systemd/system/memcached.service.
Failed to execute operation: No such file or directory


Installation has completed.
Please note details below and reboot your system

   Use a web browser to login.
      domain name: https://aaa.bbb.ccc.ddd
      username: admin
      password: bAhhwLwsvJaGQt5U

   The domain name in the browser is used by default as part of the authentication.
   If you need to login to a different domain then use username@domain.
      username: admin@aaa.bbb.ccc.ddd

   Additional information.
      https://fusionpbx.com/support.php
      https://www.fusionpbx.com
      http://docs.fusionpbx.com
\*------------------------------------------*
\* NOTE: Please save the above information. *
\* REBOOT YOUR SERVER TO COMPLETE INSTALL.  *
\*------------------------------------------*

**Comment by erxspin** [comment]
CentOS need new pre-install.sh from pull request #90

**Comment by markjcrane** [comment]
CentOS needs new pre-install.sh from pull request #90
I think it would be better to have the pre-install.sh located in centos/pre-install.sh.

Right now I'm leaning towards each install script having its own simple pre-install.sh script.

You should take a look at the changes I made to the Debian script its much easier and much cleaner. Than the old script was using the parameters. By using the config.sh its easier to customize when needed I brought the script back to its simplicity that it originally had when I made it. Simple makes it easier to maintain over time.

Also I'm not very interested in making the script interdependent. I've purposely kept them separate because I think as more operating systems are added its likely to be easier to maintain when they are not mixed together into a massive monolithic script.  So centos/resources/config.sh I like better than a shared config.sh. The reason is each operating system will have configurations that may need to be a little different to the other operating systems.

**Comment by markjcrane** [comment]
The centos script will be moved to centos/resourcs/config.sh and will migrate away from command line args.

**Comment by markjcrane** [comment]
I have another reason to keep them separate that is that I might put them each in their own repo.
Such as install-fusionpbx-bsd, install-fusionpbx-debian, install-fusionpbx-centos. Or install-bsd, install-debian, install-centos. This would allow keeping the issues and commits separated by operating system. And github pulls everything in the github repo rather than allowing you to pull only a specific directory in this case that could have been getting only the specific operating system that is being used.

**Comment by erxspin** [comment]
What you think about using primitives from #90 but using them in separate subdirs ?

Then it should be easier to port scripts to another target os with less changes and typos ...

For example use code similar to this in debian subdir - install.sh or primitives.sh

```
#dpkg/apt for Debian, Ubuntu
which apt-get && { export pkgmgr='apt-get' ; export pkgmgr_install=$pkgmgr' -y install' ; export pkgmgr_upgrade_all=$pkgmgr' update && '$pkgmgr' -y --force-yes upgrade' ; export pkgmgr_remove=$pkgmgr' -y remove' ; }
```

and use code like this in freebsd subdir

```
#pkgng/pkg for FreeBSD 10
which pkg && { export pkgmgr='pkg' ; export pkgmgr_install=$pkgmgr' install --yes' ; export pkgmgr_upgrade_all=$pkgmgr' update && '$pkgmgr' upgrade --yes' ; export pkgmgr_remove=$pkgmgr' delete --yes' ; }

```
etc ...

**Comment by markjcrane** [comment]
The whole point of the pre-install.sh is to do this.

- Update the packages if needed
- Install git
- git clone to get fusionpbx source
- start the install.sh

However starting the install.sh is debatable because it doesn't allow opportunity to customize the config.sh.

#FreeBSD Example
pkg install --yes git
cd /usr/src && git clone https://github.com/fusionpbx/fusionpbx-install.sh.git

And when you look at what is being done in the pre-install.sh its far more complex than the few simple few lines of code its supposed to run. Which makes me question if its really even useful.

**Comment by markjcrane** [comment]
You have some good notes in here that I will likely refer back to in the future.
I've updated the CentOS install script it now works with resources/config.sh.
The script is simplified a bit with more separation of concerns.

Seems to be working need to test file permissions. I kept he permission you had just needed to move some of them to finish.sh.

Thanks for your effort getting the script as far as you did that definitely helped.
